src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -152,7 +152,10 @@ object HttpGoogleServicesDAO extends GoogleServicesDAO with FireCloudRequestBuil
   def getObjectMetadata(bucketName: String, objectKey: String, authToken: String)
                     (implicit actorRefFactory: ActorRefFactory, executionContext: ExecutionContext): Future[ObjectMetadata] = {
     val request = Get( getObjectResourceUrl(bucketName, objectKey) ) ~> addCredentials(OAuth2BearerToken(authToken)) ~> sendReceive
-    request map (_.entity.asString.parseJson.convertTo[ObjectMetadata])
+    request map { response =>
+      if(response.status.isSuccess) response.entity.asString.parseJson.convertTo[ObjectMetadata]
```

Comment:
It's a shame you can't just put an `unmarshal` in the response pipeline, but I don't see another way in this case due to needing to do something different with failures.